packages/website/hooks/use-payment.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from 'react';
+
+import { userBillingSettings } from '../lib/api';
+import { plansAll } from '../components/contexts/plansContext';
+
+export const usePayment = () => {
+  /**
+   * @typedef {Object} Plan
+   * @property {string | null} id
+   * @property {string} title
+   * @property {string} description
+   * @property {string} price
+   * @property {string} base_storage
+   * @property {string} additional_storage
+   * @property {string} bandwidth
+   * @property {string} block_limit
+   * @property {string} car_size_limit
+   * @property {boolean} pinning_api
+   */
+
+  const [hasPaymentMethod, setHasPaymentMethod] = useState(false);
+  const [currentPlan, setCurrentPlan] = useState(/** @type {Plan | null} */ (null));
+
+  useEffect(() => {
```

Comment:
I think doing ajax/state management in effects like this may not scale very far. Other parts of app use [react-query](https://tanstack.com/query/v4/?from=reactQueryV3&original=https://react-query-v3.tanstack.com/)

I don't think it's important to use atm, but heads up you might find it useful in future changes